### PR TITLE
Issue #55

### PR DESCRIPTION
**Pull request by praneeth-b**
enhancement made in visualization. Episode based mean reward plot added.

### CODE REVIEW

**Comment by wkirgsn** [comment]
@praneeth-b Unit tests are failing, please rectify

**Comment by praneeth-b** [comment]
Yes.. yes.. I am working on it. I do not see any failures when i run the tests on my pc. Just a few warnings. However I feel its the last warnings that are causing the issue
[report.log](https://github.com/upb-lea/gym-electric-motor/files/4941542/report.log)

File: examples/ddpg_pmsm_dq_current_control.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def reset(self, **kwargs):
     env = gem.make(
         'emotor-pmsm-cont-v1',
         # Pass a class with extra parameters
-        visualization=MotorDashboard(plots=['i_sq', 'i_sd', 'action_1', 'action_0']), visu_period=1,
+        visualization=MotorDashboard(plots=['i_sq', 'i_sd', 'action_1', 'mean_reward']), visu_period=1,
```
[the remark this answers is not in the document]
Comment (praneeth-b) [reply examples/ddpg_pmsm_dq_current_control.py:77]: issue fixed

File: examples/pi_series_omega_control.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
         'DcSeriesCont-v1',   # replace with 'DcSeriesDisc-v1' for discrete controllers
         # Pass an instance
         #visualization=MotorDashboard(plotted_variables='all', visu_period=1),
-        visualization=MotorDashboard(plots=['omega','reward', 'i'], dark_mode=True),
+        visualization=MotorDashboard(plots=['i', 'reward', 'mean_reward'], dark_mode=False),
```
[the remark this answers is not in the document]
Comment (praneeth-b) [reply examples/pi_series_omega_control.py:19]: issue fixed

File: gym_electric_motor/visualization/motor_dashboard_plots.py [hunk 3]
```diff
@@ -365,19 +343,131 @@ def step(self, k, state, reference, action, reward, done):
...
+        super(ActionPlot, self).update()
+
+
+class EpisodeBasedPlot:
+    """Base Plot class that all episode based plots ."""
+
+    def __init__(self):
+        pass
+
+    def initialize(self, axis):
```
[the remark this answers is not in the document]
Comment (praneeth-b) [reply gym_electric_motor/visualization/motor_dashboard_plots.py:359]: bug fixed

File: gym_electric_motor/visualization/motor_dashboard_plots.py [hunk 4]
```diff
@@ -365,19 +343,131 @@ def step(self, k, state, reference, action, reward, done):
...
+    """
+    x_width = 100
+    mode = 'continuous'
+    reward_line_cfg = {
+        'color': 'blue',
+        'linestyle': '-',
+        'linewidth': 0.75,
+        'marker': 'o',
+        'markersize': 1,
+        'label': 'mean_reward'
```
[the remark this answers is not in the document]
Comment (praneeth-b) [reply gym_electric_motor/visualization/motor_dashboard_plots.py:411]: changes added

**Comment by praneeth-b** [comment]
changes made as mentioned. Please let me know if this is okay.

Screenshot from 2020-07-21 23-03-06

**Comment by wallscheid** [comment]
Nice work!

**Pull request merged**